Automattic/wp-calypso
# Add error notice for not found concierge appointment

**Built on top of https://github.com/Automattic/wp-calypso/pull/21731**

## Summary
This PR requests the appointment details for both cancellation and rescheduling flows. This way we can determine and show a proper error when the customer tries to reschedule or cancel an already cancelled appointment.

**More context 542-hg-gh**

Error when trying to cancel an already cancelled appointment: https://cloudup.com/crP1IdBWbaZ
Error when trying to reschedule a cancelled appointment: https://cloudup.com/cRBP_fsB3II

## Testing
1. Book an appointment by following the steps in https://github.com/Automattic/wp-calypso/pull/21364
2. Access the MC happiness/concierge page, under the `Lighthouse Only` tab you should see your appointment
3. Grab the id of your appointment, for now the only way to do that is by inspecting the element https://cloudup.com/c8dX8XZ2kOf and copying the `data-appointment-id` value
4. Access `http://calypso.localhost:3000/me/concierge/<business-site-slug>/<appointmentId>/cancel`
5. Cancel the appointment
6. It should show the confirmation page
7. Refresh the page and try to cancel again
8. An error should be shown
9. Change the `/cancel` at the end of the url to `/reschedule`
10. An error should be thrown because cancelled appointments cannot be found ( and rescheduled)
